File: Q1.java
import java.util.Scanner;

public class Q1 {
    public static void generatePascalTriangle(int n) {
        //write your code here
        System.out.println("Number of lines: ");
        Scanner sc = new Scanner(System.in);
        n = sc.nextInt();

        for(int i=0; i<n; i++) {
            int number = 1;

            for (int s = 1; s<n-i; s++) { //printing spaces
                System.out.print(" ");
            }
            for (int j = 0; j <= i; j++) {
                System.out.print(number + " ");

                number = number * (i-j) / (j+1);
            }
            System.out.println();
        }
    }

    public static void main(String[] args) {
        int n = 5;
        generatePascalTriangle(n);
    }
}

File: Q1.c
#include <stdio.h>

void generatePascalTriangle(int n) {
   //write your code here
   
    int i, j, k, no, value;
    printf("Enter the number of rows: ");
    scanf("%d", &no);
    
    for (i = 0; i < no; i++) {
        for (j = 0; j < (no - i - 1); j++) {
            printf(" ");
        }
        
        value = 1;
        
        for (k = 0; k <= i; k++) {
            printf("%d ", value);
            value = (value * (i - k)) / (k + 1);
        }
        
        printf("\n");
    }
}

int main() {
    int n = 5;
    generatePascalTriangle(n);
    return 0;
}
